File: problem-solving/baekjoon/2609_최대공약수와_최소공배수.go
// https://www.acmicpc.net/problem/2609
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var firstNum, secondNum int
	fmt.Fscanln(reader, &firstNum, &secondNum)

	fmt.Fprintln(writer, getGCD(firstNum, secondNum))
	fmt.Fprintln(writer, getLCM(firstNum, secondNum))
}

// 최대 공약수 구하기 (유클리드 알고리즘 이용)
func getGCD(first, second int) (gcd int) {
	if first < second { // fn에 큰 값을 오게 하기
		second, first = first, second
	}

	for second != 0 { // second가 0이 될때까지 반복
		first, second = second, first%second
	}
	return first
}

// 최소 공배수 구하기 (유클리드 알고리즘 이용)
func getLCM(first, second int) (lcm int) {
	return first * second / getGCD(first, second) // 두 수의 곱 나누기 최대공약수
}

File: problem-solving/baekjoon/2609_최대공약수와_최소공배수.py
# https://www.acmicpc.net/problem/2609
import sys

def get_gcd(first, second):
    if first < second:
        second, first = first, second
    
    while second != 0:
        first, second = second, first%second
    return first

def get_lcm(first, second):
    return first*second/get_gcd(first,second)

if __name__ == "__main__":
    first, second = map(int, sys.stdin.readline().split())
    print(int(get_gcd(first, second)))
    print(int(get_lcm(first, second)))